### DISCUSSION TITLE
Replace `cmdline` with status line fields?

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [nvim-lualine/lualine.nvim#1007] [Q&A]
Is there a way to have fields in the statusline be used for `cmdline` functionality ? 

Ex.

```
O | Search : whatisearchfor  .... | 3/12 | O
```
indead of 

```
O | ......                                        .... | .. | O
/whatisearchfor                                   3/12
```
OR 

```
O | Cmd : s/search/replace/g  .... | 3/12 | O
```

indead of 

```
O | ......                                          .... | .. | O
:s/search/replace/g
```

**Comment by yutkat** [comment]
It could be possible to create a mechanism similar to cmdheight=0 and noice.nvim, but I think it's not easy to do.
https://github.com/folke/noice.nvim

**Reply by LamprosPitsillos** [reply]
That's somewhat what i have in mind but without any extra plugins

**Reply by yutkat** [reply]
If so, it is not easy to do. cmdline is not designed for easy portability at this time.

I think it's relatively easy to simulate it by adjusting the display position of noice.nvim.
